File: kafka/project/src/main/java/com/kozitski/twit/service/KafkaWriter.java
package com.kozitski.twit.service;

import com.kozitski.twit.cli.config.KafkaConfig;
import kafka.javaapi.producer.Producer;
import kafka.producer.KeyedMessage;
import kafka.producer.ProducerConfig;
import lombok.extern.slf4j.Slf4j;

import java.util.Properties;

/**
 * The type Kafka writer.
 */
@Slf4j
public class KafkaWriter {
    private Producer<String, String> producer;

    /**
     * Instantiates a new Kafka writer.
     */
    public KafkaWriter() {
        Properties props = new Properties();
        props.put("metadata.broker.list", KafkaConfig.kafkaBroker);
        props.put("bootstrap.servers", KafkaConfig.kafkaBroker);
        props.put("acks", "all");
        props.put("delivery.timeout.ms", KafkaConfig.kafkaTimeout);
        props.put("batch.size", KafkaConfig.kafkaBatchSize);
        props.put("buffer.memory", KafkaConfig.kafkaBufferMemory);
        props.put("serializer.class", "kafka.serializer.StringEncoder");

        ProducerConfig producerConfig = new ProducerConfig(props);
        producer = new Producer<>(producerConfig);
    }

    /**
     * Write to kafka.
     *
     * @param twitterMessage the twitter message
     */
    public void writeToKafka(String twitterMessage){
        KeyedMessage<String, String> message = new KeyedMessage<>(KafkaConfig.topicName, twitterMessage);
        producer.send(message);
    }

}

File: sparkStreaming/spark_kafka-consumer/src/main/scala/com/kozitski/streaming/service/KafkaWriter.scala
package com.kozitski.streaming.service

import java.util.Properties

import com.kozitski.streaming.args.RunningArgument
import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
import org.apache.kafka.common.serialization.{StringDeserializer, StringSerializer}

/**
  * Kafka writer.
  */
class KafkaWriter() extends Serializable {
  var producer: KafkaProducer[String, String]= _

  /**
    * Set properties for kafka writer
    *
    * @param runningArgument are command line arguments [[RunningArgument]]
    */
  def init(runningArgument: RunningArgument): Unit= {

    val props = new Properties()
    props.put("bootstrap.servers", runningArgument.hostName)
    props.put("key.serializer", classOf[StringSerializer])
    props.put("value.serializer", classOf[StringSerializer])
    props.put("serializer.class", classOf[StringDeserializer])

    producer = new KafkaProducer[String, String](props)
  }

  /**
    * Write message to kafka topic
    *
    * @param topic is kafka topic name [[String]]
    * @param message is sending message [[String]]
    */
  def writeToKafkaTopic(topic: String, message: String): Unit= {
    val keyedMessage: ProducerRecord[String, String] = new ProducerRecord[String, String](topic, message)
    producer.send(keyedMessage)
  }

}
